src/ovs/aca_ovs_control.cpp

```diff
@@ -0,0 +1,1792 @@
+    void ovs_fatal_valist(int err_no, const char *format, va_list);
+    void ovs_fatal(int err_no, const char *format, ...);
+    const char *ovs_strerror(int);
+    bool str_to_uint(const char *, int base, unsigned int *);
+    inline ofp_port_t u16_to_ofp(uint16_t port);
+    void daemon_save_fd(int fd);
+    void daemonize_start(bool access_datapath);
+    int unixctl_server_create(const char *path, struct unixctl_server **);
+    void set_allowed_ofp_versions(const char *string);
+    long long int time_wall_msec(void);
+    typedef void unixctl_cb_func(struct unixctl_conn *,
+                             int argc, const char *argv[], void *aux);
+    void unixctl_command_register(const char *name, const char *usage,
+                              int min_args, int max_args,
+                              unixctl_cb_func *cb, void *aux);                          
+    void unixctl_command_reply(struct unixctl_conn *, const char *body);  
+    void unixctl_command_reply_error(struct unixctl_conn *, const char *error); 
+    void daemonize_complete(void);
+    void unixctl_server_run(struct unixctl_server *);
+    void unixctl_server_wait(struct unixctl_server *);
+    void unixctl_server_destroy(struct unixctl_server *);
+    void ofp_print_packet(FILE *stream, const void *data, size_t len,
+                 ovs_be32 packet_type);
+    const char *eth_from_hex(const char *hex, struct dp_packet **packetp);
+    void ds_put_hex_dump(struct ds *ds, const void *buf_, size_t size,
+                     uintptr_t ofs, bool ascii);
+    char *ds_steal_cstr(struct ds *);
+
+}
+
+VLOG_DEFINE_THIS_MODULE(aca);
+
+namespace aca_ovs_control
+{
+ACA_OVS_Control &ACA_OVS_Control::get_instance()
+{
+  // Instance is destroyed when program exits.
+  // It is instantiated on first use.
+  static ACA_OVS_Control instance;  
+  return instance;
+}
+
+int ACA_OVS_Control::use_names;
+int ACA_OVS_Control::verbosity;
+enum ofputil_protocol ACA_OVS_Control::allowed_protocols;
+
+int ACA_OVS_Control::control()
+{
+    int overall_rc = EXIT_SUCCESS;
+    /* -F, --flow-format: Allowed protocols.  By default, any protocol is allowed. */
+    allowed_protocols = static_cast<ofputil_protocol>(OFPUTIL_P_ANY);
+
+    char target[g_ofctl_target.size() + 1];
+    g_ofctl_target.copy(target, g_ofctl_target.size() + 1);
+    target[g_ofctl_target.size()] = '\0';
+    cout << "Target: " << g_ofctl_target << endl;
+    char options[g_ofctl_options.size() + 1];
+    g_ofctl_options.copy(options, g_ofctl_options.size() + 1);
+    options[g_ofctl_options.size()] = '\0';
+
+    if (g_ofctl_command.compare("monitor") == 0) {
+        monitor(target, options);
+    } else if (g_ofctl_command.compare("dump-flows") == 0) {
+        dump_flows(target, options);
+    } else if (g_ofctl_command.compare("packet-out") == 0) {
+        packet_out(target, options);
+    } else {
+        cout << "Usage: -c <command> -t <target> -o <options>" << endl;
```

Comment:
Do you have a logger in ACA? Let us create an issue for tracking purpose.

Reply:
we do, it is prefix with ACA_LOG_*

We will want to integrate this command line parsing into the main commandline parsing code, but that can come after this PR merged.